kategory/src/test/kotlin/kategory/laws/MonadStateLaws.kt

```diff
@@ -0,0 +1,36 @@
+package kategory
+
+import io.kotlintest.properties.forAll
+
+object MonadStateLaws {
+
+    inline fun <reified F> laws(M: MonadState<F, Int> = monadState<F, Int>(), EQ: Eq<HK<F, Int>>): List<Law> =
+            MonadLaws.laws(M, EQ) + listOf(
+                    Law("Monad State Laws: idempotence", { monadStateGetIdempotent(M, Eq.any()) }),
+                    Law("Monad State Laws: set twice", { monadStateSetTwice(M, Eq.any()) }),
+                    Law("Monad State Laws: set get", { monadStateSetTwice(M, Eq.any()) }),
+                    Law("Monad State Laws: get set", { monadStateSetTwice(M, Eq.any()) })
```

Comment:
For some HKs, like IO for the case of Monad, this Eq instance will not work as their wrapped values, their equality identity, isn't available until they're run. The Eq instance would need to be passed as a parameter.

This is okay for tests, you can see an example here: https://github.com/kategory/kategory/blob/master/kategory/src/test/kotlin/kategory/effects/IOTest.kt#L15